core/api/kotlinx-serialization-core.klib.api

```diff
@@ -0,0 +1,977 @@
+    open fun toString(): kotlin/String // kotlinx.serialization.descriptors/SerialKind.toString|toString(){}[0]
+}
+sealed class kotlinx.serialization.descriptors/StructureKind : kotlinx.serialization.descriptors/SerialKind { // kotlinx.serialization.descriptors/StructureKind|null[0]
+    constructor <init>() // kotlinx.serialization.descriptors/StructureKind.<init>|<init>(){}[0]
+    final object CLASS : kotlinx.serialization.descriptors/StructureKind // kotlinx.serialization.descriptors/StructureKind.CLASS|null[0]
+    final object LIST : kotlinx.serialization.descriptors/StructureKind // kotlinx.serialization.descriptors/StructureKind.LIST|null[0]
+    final object MAP : kotlinx.serialization.descriptors/StructureKind // kotlinx.serialization.descriptors/StructureKind.MAP|null[0]
+    final object OBJECT : kotlinx.serialization.descriptors/StructureKind // kotlinx.serialization.descriptors/StructureKind.OBJECT|null[0]
+}
+sealed class kotlinx.serialization.modules/SerializersModule { // kotlinx.serialization.modules/SerializersModule|null[0]
+    abstract fun <#A1: kotlin/Any> getContextual(kotlin.reflect/KClass<#A1>, kotlin.collections/List<kotlinx.serialization/KSerializer<*>> =...): kotlinx.serialization/KSerializer<#A1>? // kotlinx.serialization.modules/SerializersModule.getContextual|getContextual(kotlin.reflect.KClass<0:0>;kotlin.collections.List<kotlinx.serialization.KSerializer<*>>){0§<kotlin.Any>}[0]
+    abstract fun <#A1: kotlin/Any> getPolymorphic(kotlin.reflect/KClass<in #A1>, #A1): kotlinx.serialization/SerializationStrategy<#A1>? // kotlinx.serialization.modules/SerializersModule.getPolymorphic|getPolymorphic(kotlin.reflect.KClass<in|0:0>;0:0){0§<kotlin.Any>}[0]
+    abstract fun <#A1: kotlin/Any> getPolymorphic(kotlin.reflect/KClass<in #A1>, kotlin/String?): kotlinx.serialization/DeserializationStrategy<#A1>? // kotlinx.serialization.modules/SerializersModule.getPolymorphic|getPolymorphic(kotlin.reflect.KClass<in|0:0>;kotlin.String?){0§<kotlin.Any>}[0]
+    abstract fun dumpTo(kotlinx.serialization.modules/SerializersModuleCollector) // kotlinx.serialization.modules/SerializersModule.dumpTo|dumpTo(kotlinx.serialization.modules.SerializersModuleCollector){}[0]
+    constructor <init>() // kotlinx.serialization.modules/SerializersModule.<init>|<init>(){}[0]
+    final fun <#A1: kotlin/Any> getContextual(kotlin.reflect/KClass<#A1>): kotlinx.serialization/KSerializer<#A1>? // kotlinx.serialization.modules/SerializersModule.getContextual|getContextual(kotlin.reflect.KClass<0:0>){0§<kotlin.Any>}[0]
+}
+// Targets: [native, wasmJs, wasmWasi]
+final class <#A: kotlin/Any?, #B: kotlin/Any?> kotlinx.serialization.internal/LinkedHashMapSerializer : kotlinx.serialization.internal/MapLikeSerializer<#A, #B, kotlin.collections/Map<#A, #B>, kotlin.collections/HashMap<#A, #B>> { // kotlinx.serialization.internal/LinkedHashMapSerializer|null[0]
```

Comment:
@fzhinkin PTAL at this declaration. It is located in common source (https://github.com/Kotlin/kotlinx.serialization/blob/a27c36594ce4f6ef3a6f869cf38f5fa9ff9bd71e/core/commonMain/src/kotlinx/serialization/internal/CollectionSerializers.kt#L253) yet it is treated as platform-specific in two versions: one for JS and one for all other targets. Probably because supertype list is different:
`kotlinx.serialization.internal/MapLikeSerializer<#A, #B, kotlin.collections/Map<#A, #B>, kotlin.collections/HashMap<#A, #B>>` vs `kotlinx.serialization.internal/MapLikeSerializer<#A, #B, kotlin.collections/Map<#A, #B>, kotlin.collections/LinkedHashMap<#A, #B>>`. Do we not have LinkedHashMap in Native?

I'm not sure something actionable can be done here, but still it worth attention.

Reply:
For native & wasm, [LinkedHashMap is a typealias to HashMap](https://github.com/JetBrains/kotlin/blob/1242d38fcfd06ac7e20d5b920cb4d275acda4f09/libraries/stdlib/native-wasm/src/kotlin/collections/HashMap.kt#L774), so the actual type differs from JS's.